src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -537,10 +547,10 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                                 completionContext.ReplacementLength = replacementLength = 0;
                                 result = GetResultForAttributeArgument(completionContext, ref replacementIndex, ref replacementLength);
                             }
-                            else
+                            else if (lastAst.Parent is DynamicKeywordStatementAst || lastAst.Parent is CommandExpressionAst)
```

Comment:
We are making two special checks here. Maybe we should only have one special check, like the following. What do you think?

```c#
if (lastAst is AttributeAst)
{
    // the original attribute handling
}
else if (lastAst is HashtableAst hashTableAst &&
         !(lastAst.Parent is DynamicKeywordStatementAst) &&
         CheckForPendingAssignment(hashTableAst))
{
    return null;
}
else
{
    // the original else block
}
```

Reply:
Looks good. Thanks!

Fixed.